src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -260,21 +260,24 @@ inline void JavaThread::set_done_attaching_via_jni() {
 }
 
 inline bool JavaThread::is_exiting() const {
-  // Use load-acquire so that setting of _terminated by
-  // JavaThread::exit() is seen more quickly.
   TerminatedTypes l_terminated = Atomic::load_acquire(&_terminated);
-  return l_terminated == _thread_exiting || check_is_terminated(l_terminated);
+  return l_terminated == _thread_exiting ||
+         l_terminated == _thread_gc_barrier_detached ||
+         check_is_terminated(l_terminated);
+}
+
+inline bool JavaThread::is_oop_safe() const {
```

Comment:
Do we want to document, and assert, that this is only intended for use by the current thread on itself? I can't imagine any scenario where we would need to ask whether an arbitrary thread is oop-safe. ??

Reply:
@dholmes-ora - Thanks very much for the re-review.

I'll accept your "good as-is" and proceed with integration.

I think I've tweaked this fix enough at this point. Especially if we end up replacing
it with support in the GC subsystems instead.